common/transport/transport-api/src/main/java/org/thingsboard/server/common/transport/service/DefaultTransportService.java

```diff
@@ -731,7 +731,9 @@ public void process(TransportProtos.SessionInfoProto sessionInfo, TransportProto
 
     @Override
     public void recordActivity(TransportProtos.SessionInfoProto sessionInfo) {
-        recordActivityInternal(sessionInfo);
+        if(sessionInfo != null) {
```

Comment:
Move the check to recordActivityInternal

Reply:
Add a WARN log in case it is null (since we will add additional check isProvisionOnly above):
log.warn("Session info is missing")